src/mnelab/dialogs/filter.py

```diff
@@ -3,34 +3,107 @@
 # License: BSD (3-clause)
 
 from PySide6.QtWidgets import (
+    QButtonGroup,
     QDialog,
     QDialogButtonBox,
     QGridLayout,
+    QHBoxLayout,
     QLabel,
     QLineEdit,
+    QRadioButton,
     QVBoxLayout,
 )
 
 
 class FilterDialog(QDialog):
-    def __init__(self, parent):
+    def __init__(self, parent=None):
         super().__init__(parent)
         self.setWindowTitle("Filter data")
         vbox = QVBoxLayout(self)
-        grid = QGridLayout()
-        grid.addWidget(QLabel("Low cutoff frequency (Hz):"), 0, 0)
+
+        # radio buttons
+        self.bandpass_button = QRadioButton("Bandpass")
+        self.bandpass_button.setChecked(True)
+        self.lowpass_button = QRadioButton("Lowpass")
+        self.highpass_button = QRadioButton("Highpass")
+        self.notch_button = QRadioButton("Notch")
+
+        self.filter_group = QButtonGroup(self)
+        self.filter_group.addButton(self.bandpass_button)
+        self.filter_group.addButton(self.lowpass_button)
+        self.filter_group.addButton(self.highpass_button)
+        self.filter_group.addButton(self.notch_button)
```

Comment:
Can you put these two `QButtonGroup`s into two `QGroupBox` containers? That way, the dialog would be a bit more readable, because you can add a title for each `QGroupBox` (e.g., "Filter type" and "Filter Settings").